### PR TITLE
Some changes to New-OctopusArtifact to allow pipelining in a list

### PR DESCRIPTION
**Pull request by stopthatastronaut**
I wanted to do `Get-ChildItem $folder | New-OctopusArtifact`. The Cmdlet didn't support it.

So I tweaked the code a little. Now it supports pipelining in a list of files.

I also wrote a few pester tests around this so I could test locally, but I was really unsure how to slot them in. Can add those if wanted

### CODE REVIEW

**Comment by matt-richardson** [comment]
Dont forget to:
a) link it to an issue in OctopusDeploy/Issues
b) update Tentacle and Server(?) with the new package

File: source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1 [hunk 1]
```diff
@@ -265,29 +265,39 @@ function Fail-Step([string] $message)
...
+
+	    $length = ([System.IO.FileInfo]$fullpath).Length;
+	    if (!$length) {
+		    $length = 0;
+	    }
+	    $length = Convert-ServiceMessageValue($length.ToString());
+
+	    $fullpath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($fullpath)
+	    $fullpath = [System.IO.Path]::GetFullPath($fullpath)
+        $servicepath = Convert-ServiceMessageValue($fullpath)
```
Comment (matt-richardson) [inline source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1]: Can we fix the indenting at the same time?

**Comment by stopthatastronaut** [comment]
One bit of the build chain is failing - @alfhenrik  reckons that's nothing to do with this change, because it's on the Linux/mono path. If nobody objects, I'll merge this later today

File: source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1 (hunk 1 above)
Comment (stopthatastronaut) [reply source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1]: I'll push one more commit to do that

**Pull request merged**